### ISSUE TITLE
Bug: VSCode v1.78.2 does not show badge/color for modified files in explorer

### ISSUE THREAD

**Issue by amirhessam88** [microsoft/vscode#184293]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.78
- OS Version: Apple M1 Ventura 13.4

Steps to Reproduce:

1. Changed files show up in `source control` tabs; however, they dont show up in the `explorer` tab while I have the followings in my `settings`
```
    "explorer.decorations.badges": true,
    "explorer.decorations.colors": true,

```
2. I did see similar issues but none of them has a real solution. I am wondering if y'all can help me to figure out what should I do?

The image below shows the current behavior
![image](https://github.com/microsoft/vscode/assets/18218820/753392d0-395b-4462-9915-d61fb76cd857)

I did also ran `extension bisecting` to see whether this is an issue with some extensions; however, it was not
<img width="1719" alt="image" src="https://github.com/microsoft/vscode/assets/18218820/df6daec9-5546-4873-9d08-d2a1c4cc4f79">
<img width="244" alt="image" src="https://github.com/microsoft/vscode/assets/18218820/d63a096c-5c72-48f6-abe7-15c492849131">

**Comment by VSCodeTriageBot** [comment]
Thanks for creating this issue! It looks like you may be using an old version of VS Code, the latest stable release is 1.78.2. Please try upgrading to the latest version and checking whether this issue remains.

Happy Coding!

**Comment by amirhessam88** [comment]
I have installed `1.78.2`; the issue still remains

**Comment by amirhessam88** [comment]
git@github.com:amirhessam88/my-dv-pal.git

**Comment by amirhessam88** [comment]
![image](https://github.com/microsoft/vscode/assets/18218820/9042713e-af44-46b9-9db4-1d626eaf978b)

**Comment by amirhessam88** [comment]
similar issues without any resolutions:
- https://github.com/microsoft/vscode/issues/176749

**Comment by amirhessam88** [comment]
this seems related to this PR https://github.com/microsoft/vscode/pull/105656

**Comment by amirhessam88** [comment]
CC @joaomoreno

**Comment by lramos15** [comment]
Did this work in previous versions? Is there something special about your file structure?

**Comment by amirhessam88** [comment]
@lramos15 Thanks for the response; I had an old version of vscode (prolly ~Sep2021 release) and upgraded my system and installed the latest one. I noticed the issue is related to the way `code .` works. Apparently, I did not have the `code` command installed on my path and I was getting away with `code () { VSCODE_CWD="$PWD" open -n -b "com.microsoft.VSCode" --args $* ;}`; 
I learned this when I open the project via folder rather the command line. After that everything is working as expected.
